docs/core/diagnostics/diagnostics-in-containers.md

```diff
@@ -0,0 +1,69 @@
+
+## Using .NET Core global CLI tools in a container
+
+The .NET Core global CLI diagnostic tools ([`dotnet-counters`](dotnet-counters.md), [`dotnet-dump`](dotnet-dump.md), [`dotnet-gcdump`](dotnet-gcdump.md), and [`dotnet-trace`](dotnet-trace.md)) are designed to work in a wide variety of environments and should all work directly in Docker containers. Because of this, these tools are the preferred method of collecting diagnostic information for .NET Core scenarios targeting .NET Core 3.0 or above (or 3.1 or above in the case of `dotnet-gcdump`) in containers.
+
+The only complicating factor of using these tools in a container is that they are installed with the .NET Core SDK and many Docker containers run without the .NET Core SDK present. One easy solution to this problem is to install the tools in the initial Docker image. The tools don't need the .NET Core SDK to run, only to be installed. Therefore, it's possible to create a Dockerfile that installs the tools in a build stage (where the .NET Core SDK is present) and then copies the binaries into the final image. The only downside to this approach is increased Docker image size (each tool is roughly 100 KB, give or take, depending on the specific tool and OS).
+
+```dockerfile
+# In build stage
+# Install desired .NET CLI diagnostics tools
+RUN dotnet tool install --tool-path /tools dotnet-trace
+RUN dotnet tool install --tool-path /tools dotnet-counters
+RUN dotnet tool install --tool-path /tools dotnet-dump
+
+...
+
+# In final stage
+# Copy diagnostics tools
+WORKDIR /tools
+COPY --from=build /tools .
+```
+
+Alternatively, the .NET Core SDK can be installed in a container when needed in order to install the CLI tools. Be aware that installing the .NET Core SDK will have the side-effect of reinstalling the .NET Core runtime. So be sure to install the version of the SDK that matches the runtime present in the container.
+
+### Using .NET Core global CLI tools in a sidecar container
+
+If you would like to use .NET Core global CLI diagnostic tools to diagnose processes in a different container, bear the following additional requirements in mind:
+
+1. The containers must, of course, [share a process namespace](https://docs.docker.com/engine/reference/run/#pid-settings---pid) (so that tools in the sidecar container can access processes in the target container).
+2. The .NET Core global CLR diagnostic tools need access to files the .NET Core runtime writes to the /tmp directory, so the /tmp directory must be shared between the target and sidecar container via a volume mount. If you attempt to use the diagnostic tools from a sidecar container without sharing the /tmp directory, you will get an error about the process "not running compatible .NET runtime."
```

Comment:
Forcing all CLR containers to mount a common `/tmp` directory seems messy/risky.  I could guess this is the side car mounting the CLR container's `/tmp` directory, but that seems to imply a nested container.  If that is the case the description is not very clear.

Sample sessions/scripts might be in order here and for the samples below.

Reply:
I think a simple k8s manifest or pair of docker cli commands would work here.  For example, something like

```
docker run --rm -v scratch:/tmp appcontainer
```

```
docker run -it -v scratch:/tmp sidecar bash
```

would probably be sufficient.

Reply:
Nit: you'll need to create the volume using
```
docker volume create scratch
```

Reply:
If this is a docker server running on my dev box and `scratch` is a local directory, wouldn't the above be sufficient?

Reply:
Assuming you have a local directory by that name and you've granted the docker daemon access to that volume.
Since the domain socket is ephemeral, there's not much value in sharing it with the host.

I found creating a volume is just more reliable for this scenario

Reply:
I'll add Docker commands or K8s manifest snippets to try make this clearer.